agent/consul/leader.go

```diff
@@ -1332,11 +1332,12 @@ AFTER_CHECK:
 			Status:  api.HealthPassing,
 			Output:  structs.SerfCheckAliveOutput,
 		},
```

Comment:
what does means the removal of `SkipNodeUpdate: true` ?
To encore the Healthy on Serf?

Reply:
@kyhavlov can correct me if I am wrong but I believe the reasoning here is that we may want to update the node id or address. In that case using SkipNodeUpdate wouldn't update either of those two fields.

The few lines added below copy in the nodes tagged addresses and node meta of the existing registration so that we don't clobber those (as we cannot get them via serf).

One question I have is whether their is a race condition here. We are pulling the tagged addresses and node meta of the node out of a snapshot of the fsm. Wouldn't it be possible for anti entropy on the node to kick in and update the tagged addresses or node meta between when we grab it and when we finally get raft to apply the change?

Reply:
The SkipNodeUpdate field prevents the node state from being written if it doesn't already exist (which we don't want since we're updating the ID here): https://github.com/hashicorp/consul/blob/master/agent/consul/state/catalog.go#L309

With regard to the possibility of a race condition, anti-entropy starts on a delay where this handleAliveMember function gets triggered immediately by the node joining, so in practice that shouldn't happen. Even if anti-entropy did update the meta/tagged addresses in the middle of this (and get clobbered by the leader's update) it should set them back to the updated values afterward when it kicks in again.